hack/cmd/mco-builder/internal/builders/podman.go

```diff
@@ -0,0 +1,84 @@
+package builders
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/openshift/machine-config-operator/hack/internal/pkg/errors"
+	"k8s.io/klog/v2"
+)
+
+type podmanBuilder struct {
+	opts Opts
+}
+
+func newPodmanBuilder(opts Opts) Builder {
+	return &podmanBuilder{opts: opts}
+}
+
+func (p *podmanBuilder) Build() error {
+	if err := validateLocalBuilderMode(p.opts); err != nil {
+		return err
+	}
+
+	if err := p.buildContainer(); err != nil {
+		return fmt.Errorf("unable to build container: %w", err)
+	}
+
+	return nil
+}
+
+func (p *podmanBuilder) Push() error {
+	if err := p.tagContainerForPush(); err != nil {
+		return fmt.Errorf("could not tag container: %w", err)
+	}
+
+	if err := p.pushContainer(); err != nil {
+		klog.Info("Push failed, falling back to Skopeo")
+		return pushWithSkopeo(p.opts, BuilderTypePodman)
+	}
+
+	return nil
+}
+
+func (p *podmanBuilder) tagContainerForPush() error {
+	cmd := exec.Command("podman", "tag", localPullspec, p.opts.FinalPullspec)
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return errors.NewExecError(cmd, out, err)
+	}
+
+	return nil
+}
+
+func (p *podmanBuilder) buildContainer() error {
+	podmanOpts := []string{"build", "-t", localPullspec, "--network", "slirp4netns", "--jobs", "3", "--file", p.opts.DockerfileName, p.opts.RepoRoot}
```

Comment:
thought (non-blocking): I've been toying with an abstraction for complex calls to binaries such as `podman` that would basically allow you to do something like:

```golang
type PodmanBuild struct {
    Tag string
    Jobs int
    ContainerfilePath string
    BuildContextPath string
    AuthFilePath string
}

// Then you could do something like:

func main() {
    pb := &PodmanBuild{
        Tag: "quay.io/org/repo:tag",
        Jobs: 3,
        ContainerfilePath: "/path/to/mco/repo/Containerfile",
        BuildContextPath: "/path/to/mco/repo",
    }

    if authFilePath != "" {
        pb.AuthFilePath = authFilePath
    }

    if err := pb.Build(); err != nil {
        panic(err)
    }
}
```

I'm just putting the thought here for posterity. I'm not expecting you to implement it or anything :smile: